### PR TITLE
Add collection methods to `dask_histogram.boost.Histogram` class

### LINKED ISSUE #36: bug: dask histograms are not recognized as `compute`-able
if I have a dict of stuff to compute like:
```python3
out = {'mass': Histogram(
   StrCategory(['ZJets', 'Data'], growth=True),
   Regular(30000, 0.25, 300),
   storage=Double()) # (has staged fills),
 'pt': Histogram(
   StrCategory(['ZJets', 'Data'], growth=True),
   Regular(30000, 0.24, 300),
   storage=Double()) # (has staged fills),
 'cutflow': {'ZJets_pt': dask.awkward<sum-agg, type=Scalar, dtype=Unknown>,
  'ZJets_mass': dask.awkward<sum-agg, type=Scalar, dtype=Unknown>,
  'Data_pt': dask.awkward<sum-agg, type=Scalar, dtype=Unknown>,
  'Data_mass': dask.awkward<sum-agg, type=Scalar, dtype=Unknown>}}
 ```
where mass and pt are dask_histograms. When I do dask.compute(out) the dask_awkward sums are computed but not the histograms!

and in fact:
```python3
dask.compute(out["mass"])
```
just returns again the dask_histogram as though it's not recognized as a dask collection at all.

### PR DESCRIPTION
**Pull request by douglasdavis**
@lgray I'm hoping this will fix #36. Still need to add tests but wanted to at least make this available for you to experiment with

### CODE REVIEW

**Comment by lgray** [comment]
Ah yeah - I had patched most of these in locally and it seems to work!

As a side question is adding dask-based adding of boost histograms supported?

**Comment by lgray** [comment]
Ah - I see what's missing for that.

**Comment by douglasdavis** [comment]
Yea we'll need to add `__add__`, etc. as well

**Comment by lgray** [comment]
I also figured out part of the issue with string/int category fills. It's actually more an issue in dask_awkward.
The first error in the chain is that dask_awkward *always* assumes args[0] is `dask_awkward.Array` and the most usual idiom for categorical axes in histograms is to have those first!

File: src/dask_histogram/boost.py [hunk 1]
```diff
@@ -200,35 +210,14 @@ def fill(  # type: ignore
...
-            return self
-        if not self.empty():
-            result_view = self.view(flow=True) + self._staged.compute().view(flow=True)
+            self._staged = new_fill
         else:
-            result_view = self._staged.compute().view(flow=True)
-        self[...] = result_view
-        self._staged = None
-        return self
+            self._staged += self._staged
```
Comment (lgray) [inline src/dask_histogram/boost.py:216]: shouldn't this be `+= new_fill`? How is this not caught in the tests?

Comment (douglasdavis) [reply src/dask_histogram/boost.py:216]: Indeed, thanks for catching! need to add another test that does multliple fills

**Pull request merged**